src/test/StkAaveRetrievalE2E.t.sol

```diff
@@ -0,0 +1,113 @@
+        vm.prank(AAVE_WHALE);
+        proposalId = DeployMainnetProposal._deployMainnetProposal(
+            address(proposalPayload),
+            0x344d3181f08b3186228b93bac0005a3a961238164b8b06cbb5f0428a9180b8a7 // TODO: Replace with actual IPFS Hash
+        );
+    }
+
+    function testClaimerUnsetPreProposal() public {
+        // Call retrieve() on StkAaveRetrieval contract as Balancer multisig without the Proposal being executed
+        vm.prank(balancerMultisig);
+        vm.expectRevert(bytes("Contract not set as claimer"));
+        stkAaveRetrieval.retrieve();
+    }
+
+    function testRetrieveNotBalancerMultisigPreProposal() public {
+        // Call retrieve() on StkAaveRetrieval contract without the Proposal being executed
+        // the msg.sender will be this contract, which is NOT the balancer multisig
+        GovHelpers.passVoteAndExecute(vm, proposalId);
+        vm.expectRevert(bytes("Only Balancer Multisig"));
+        stkAaveRetrieval.retrieve();
+    }
+
+    function testExecute() public {
+        // check that originally claimer is unset, then is correctly set to retrieval contract
+        IAaveIncentivesController incentivesController = IAaveIncentivesController(incentivesControllerAddr);
+        assertEq(incentivesController.getClaimer(balancerDAO), address(0), "CLAIMER_NOT_ZERO");
+        // Pass vote and execute proposal
+        GovHelpers.passVoteAndExecute(vm, proposalId);
+        assertEq(
+            incentivesController.getClaimer(balancerDAO),
+            address(stkAaveRetrieval),
+            "CLAIMER_NOT_RETRIEVAL_CONTRACT"
+        );
+    }
+
+    function testRetrievePostProposal() public {
+        GovHelpers.passVoteAndExecute(vm, proposalId);
+
+        // check stkAaveBalance of balancermultisig is zero
+        assertEq(STK_AAVE.balanceOf(balancerMultisig), 0, "BALANCER_MULTISIG_STK_AAVE_BALANCE_NOT_ZERO");
+
+        // check initial aToken reward balances of Balancer DAO
+        uint256 before_balance_ADAI = wrapped_aDAI.getUnclaimedRewards(balancerDAO);
+        uint256 before_balance_AUSDC = wrapped_aUSDC.getUnclaimedRewards(balancerDAO);
+        uint256 before_balance_AUSDT = wrapped_aUSDT.getUnclaimedRewards(balancerDAO);
+        assertTrue(before_balance_ADAI > 0, "ZERO_WRAPPED_ADAI_REWARDS_INITIAL_BALANCE");
+        assertTrue(before_balance_AUSDC > 0, "ZERO_WRAPPED_AUSDC_REWARDS_INITIAL_BALANCE");
+        assertTrue(before_balance_AUSDT > 0, "ZERO_WRAPPED_AUSDT_REWARDS_INITIAL_BALANCE");
+
+        uint256 beforeStkAAVEBalance = STK_AAVE.balanceOf(balancerMultisig);
+        // Mock as Balancer Multisig and call retrieve() on StkAaveRetrieval contract
+        vm.prank(balancerMultisig);
+        stkAaveRetrieval.retrieve();
+
+        // check that stkAave balance is correct now
+        uint256 afterStkAAVEBalance = STK_AAVE.balanceOf(balancerMultisig);
+        assertTrue(afterStkAAVEBalance > beforeStkAAVEBalance, "BALANCER_MULTISIG_STK_AAVE_BALANCE_UNCHANGED");
```

Comment:
These assertions are fine. However is there any way to calculate what the actual/exact stkAAVE balance would be after executing the retrieval? 

That would be the perfect test if you can simulate those calculations here using the before balance of aDAI, aUSDC and aUSDT and show that the the value you're getting is infact the same as the stkAAVE balance after executing the retrieval.

Reply:
This is what I had before with the amounts hardcoded in. I'm a bit confused, can you clarify?

Reply:
But he claim rewards function doesn't have an amount, it's all or none
`IStaticATokenLM(WRAPPED_TOKENS[i]).claimRewardsOnBehalf(BALANCER_DAO, BALANCER_MULTISIG, true);`

meaning there's no `amountToBeTransferred`

Reply:
Unless in abstract terms we're ok with it being sum(unclaimed amount on ADAI, unclaimed amount onAUSDC, unclaimed amount on AUSDT)

Reply:
That seems to work

Reply:
Yup. that's what I meant.

Reply:
Looks like it's solved now.